src/nvim/eval.c

```diff
@@ -24033,6 +24050,7 @@ static inline TerminalJobData *common_job_init(char **argv,
   }
   Process *proc = (Process *)&data->proc;
   proc->argv = argv;
+  proc->quote_argv = quote_argv;
```

Comment:
So the quoting would be disabled if `shellxquote` was not set. That is an interesting choice. Essentially using shellxquote to make this behaviour paremeterizable.

I cannot do it for the general case (in that line), because it would affect `jobstart([])`/`system([])`/etc and the goal of this PR is only to change the behaviour of the string variants `system('...')`/`jobstart('...')` - it makes sense for shellxquote to affect the later but not the former.

Could do it earlier as `!shellcmd  || p_sxq == NULL`